WowCharComparerWebApp/Controllers/User/RegisterController.cs

```diff
@@ -76,24 +88,71 @@ public bool CheckUsername(string accountName, ComparerDatabaseContext db)
         {
-            return email.Contains("@") && email.Contains(".");
+            try
+            {
+                MailAddress mail = new MailAddress(email);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                //TODO write to user that mail format is incorrect or already is in database
+                return false;
+            }
         }
 
-        public void SendVerificationMail(string userEmail)
+        public void SendVerificationMail(string userEmail, Guid activationGuid)
         {
+            var test = HttpContext.Request.Path;
+
             SmtpClient client = new SmtpClient("smtp.gmail.com", 587)
             {
                 DeliveryMethod = SmtpDeliveryMethod.Network,
                 UseDefaultCredentials = false,
-                Credentials = new NetworkCredential(APIConf.WowCharacterComparerEmail, APIConf.WoWCharacterComparerEmailPassword)
+                Credentials = new NetworkCredential(APIConf.WowCharacterComparerEmail, APIConf.WoWCharacterComparerEmailPassword),
+                Timeout = 100000,
+                EnableSsl = true             
             };
 
-            MailMessage msg = new MailMessage();
+            string protocol = HttpContext.Request.IsHttps ? "https" : "http";
+
+            string activationLink = $"{protocol}://www.{ HttpContext.Request.Host}/Register/AccountConfirmation";
+
+            MailMessage msg = new MailMessage();    
             msg.To.Add(userEmail);
             msg.From = new MailAddress(APIConf.WowCharacterComparerEmail);
-            msg.Subject = "Verification";
-            msg.Body = "This is a email to verification your World of Warcraft Character Comparer account";
+            msg.IsBodyHtml = true;
+            msg.Subject = "Verification";          
+            msg.Body = "<p> This is a email to verification your World of Warcraft Character Comparer account.</p>" +
+                       $"<p> Activation code: <b> {activationGuid.ToString()} </b> </p>" + 
+                       "Please active your account " + $"<a href=\"{activationLink}\">here </a>";
             client.Send(msg);
         }
+
+        [Route("Register/ConfirmUserAccount")]
+        [HttpPost]
+        public IActionResult ConfirmUserAccount(string userAccountActivationToken)
+        {
+            try
+            {
+                using (ComparerDatabaseContext db = new ComparerDatabaseContext())
+                using (IDbContextTransaction transaction = db.Database.BeginTransaction())
+                {
+                    if (db.Users.Select(x => x.VerificationToken.ToString().ToUpper()).ToList().Contains(userAccountActivationToken.ToUpper()))
+                    {
+                        var verificationToken = new SqlParameter("@token", userAccountActivationToken);
+                        db.Database.ExecuteSqlCommand("UPDATE dbo.Users SET Verified = 1 WHERE VerificationToken =@token", verificationToken);
```

Comment:
Use entity framework instead of ExecuteSqlCommand!

Reply:
Refactored using Enity Framework